Libraries/acdriver/Sources/Extensions.cpp

```diff
@@ -0,0 +1,73 @@
+/**
+ Copyright (c) 2015-present, Facebook, Inc.
+ All rights reserved.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree. An additional grant
+ of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#include <acdriver/Extensions.h>
+#include <libutil/Options.h>
+
+using acdriver::Extensions;
+
+static std::pair<bool, std::string>
+InsertNextImageType(Extensions::ImageTypeSet *imageTypeSet, std::vector<std::string> const &args, std::vector<std::string>::const_iterator *it)
+{
+    ext::optional<std::string> value;
+    auto ret = libutil::Options::Next<std::string>(&value, args, it);
+    if (!ret.first) {
+        return ret;
+    }
+    ext::optional<Extensions::ImageType> type = Extensions::ImageTypeFromFileExtension(*value);
+    if (!type) {
+        return std::make_pair(false, "unknown image type " + *value);
+    }
+    imageTypeSet->insert(*type);
+    return std::make_pair(true, std::string());
+}
+
+std::pair<bool, std::string> Extensions::ActoolOptions::
+parseArgument(std::vector<std::string> const &args, std::vector<std::string>::const_iterator *it)
+{
+    std::string const &arg = **it;
+
+    if (arg == "--allow-extensions") {
+        return libutil::Options::Current<bool>(&_allowExtensions, arg);
+    } else if (arg == "--allow-image-types") {
+        return InsertNextImageType(&_allowImageTypes, args, it);
+    } else {
```

Comment:
Nit: it's not the extension's responsibility decide what happens if an argument can't be parsed.  Make this parseArgument return a nullable, check for it in the main options file, and return the appropriate thing there.